### PR TITLE
mesh-vpn: fully abstract VPN methods

### PR DESCRIPTION
**Pull request by blocktrron**
This fully abstracts VPN methods, making gluon-mesh-vpn-fastd and
gluon-mesh-vpn-tunneldigger completely self-contained.

Provide a LUA interface for generic interacting with VPN methods in
gluon-mesh-vpn-core and web packages.

This also adds the ability to install tunneldigger and fastd to the same
image, selecting the VPN method based on the selected domain.

### CODE REVIEW

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua:10]: Maybe we should find another term here, as "method" is already used in fastd configuration (and also appears in site.conf for this reason).

**Comment by NeoRaider** [comment]
I see a few conceptual issues that need to be discuss:
- Should the chosen VPN implementation only be a matter of site/domain config, or should is be possible for the user to change this (for example turning the security/performance mode of `gluon-web-mesh-vpn-fastd` into a generic switch hat selects between fastd and tunneldigger)?
- There are a few places in this PR that query a specific method whether it is enabled. This doesn't make sense to me - VPN enable status should be a single value in `/etc/config/gluon`. The `enabled` flags of the individual VPN implementation config files are dependent values that are populated based on the global enable status and the selected VPN implementation; they should only be set, but never be queried by Gluon's scripts (except for migration purposes, in case the global enable status doesn't exist yet)

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua:21]: Just pass `true` or `false` to `uci:set`, `simple-uci` will turn that into `'0'` and `'1'`.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua:69]: Indentation should use tabs

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua:38]: Missing final newline in multiple files

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua:10]: How about `vpn_proto`?

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua:21]: Missed this. Fixed.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua:69]: Fixed.

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua]: Fixed.

**Comment by blocktrron** [comment]
@NeoRaider 

For the first point, I would propose to leave it to the site configuration for now. The goal of this PR is not to add this functionality but to remove the core VPN package dependencies on the specific VPN implementation packages.

Your second remark is correct. I've simply didn't correctly interpret the context and it's goal for which the specific VPN states are extracted. I'll fix that.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua
Comment (mweinelt) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua]: Protocol is the correct identifier if you want to differentiate between fastd, tunneldigger and wireguard. They're different protocols.

Comment (NeoRaider) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua]: I'm thinking about adding L2TP support to fastd though, making it an alternative to tunneldigger...

I think a more general term would be something like "feature provider" or "function provider" - unfortunately this could get mixed up with service providers :/

File: package/gluon-mesh-vpn-core/luasrc/lib/gluon/mesh-vpn/update-config
Comment (NeoRaider) [inline package/gluon-mesh-vpn-core/luasrc/lib/gluon/mesh-vpn/update-config:46]: You are removing the code that disables the unavailable VPNs, this is not correct.

Comment (NeoRaider) [reply package/gluon-mesh-vpn-core/luasrc/lib/gluon/mesh-vpn/update-config:46]: Hmm, I wrote the code this way so that even VPNs that are not installed anymore get disabled (for the case when for example `fastd` is installed, but `gluon-mesh-vpn-fastd` isn't).

What we really want is a generic facility that allows removed packages to clean up after themselves on firmware upgrades, that way we wouldn't need a full list of packages that could exist. I have a few ideas how to solve this...

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua:14]: The impl-specific `enabled()` methods can be removed.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua:43]: As `get_active_method_name()` needs to load the method modules anyways, it would be nicer to merge it into `get_active_method()` (which can then have two return values, the name and the implementation).

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn/core.lua:77]: Again, `get_inactive_methods()` and `get_inactive_method_names()` both iterate over all methods and load all impls. Let's have a single `get_inactive_methods()` function that returns a table of impls keyed by the names.

Also, this is currently unused, but I assume it might become used in the future?

File: package/gluon-config-mode-mesh-vpn/luasrc/lib/gluon/config-mode/wizard/0300-mesh-vpn.lua
Comment (NeoRaider) [reply package/gluon-config-mode-mesh-vpn/luasrc/lib/gluon/config-mode/wizard/0300-mesh-vpn.lua:71]: Maybe we should first get rid of all these useless return values first and simply do a global `uci commit` at the end of the config wizard... It will only write files that need to change anyways.

**Review by NeoRaider — CHANGES_REQUESTED** [review]

**Comment by CodeFetch** [comment]
@NeoRaider
> Should the chosen VPN implementation only be a matter of site/domain config, or should is be possible for the user to change this (for example turning the security/performance mode of gluon-web-mesh-vpn-fastd into a generic switch hat selects between fastd and tunneldigger)?

I've been working on IPSec using L2TP where I'd like to be able to let the user configure the encryption this way. Please make it as generic as possible or we will rip it out again, later.

**Comment by blocktrron** [comment]
Fixed all outstanding comments.

Regarding the teardown of old VPN methods - I tend to leave it as - is for now. This might not be perfect, but we remain consistent in the behavior.

For the future, we can have a teardown manager which manages teardown in case a package does no exist anymore. The goal of allowing a third-party VPN method from a community feed is still possible this way, they have to take care of teardown themselves though. 

@AiyionPrime Not build / ru ntested yet although the linter is happy.

**Comment by AiyionPrime** [comment]
Thanks!
I just compiled and installed it on one of my cudy wr1000 (ramips-mt76x8).
http://[2001:678:978:114:b64b:d6ff:fe22:4ebc]/cgi-bin/status

You five got access to it, welcome to my kitchen; obviously leave the rest of the network untouched.

**Comment by AiyionPrime** [comment]
I rebuilt the firmware;  http://meshvpnabstracttest.n.ffh.zone/cgi-bin/status runs now on 251d550eed8290a78ecadcea2f5326d436e4b95f.

In case our dns is mesed up:
http://[2001:678:978:114:b64b:d6ff:fe22:4ebc]/cgi-bin/status

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (AiyionPrime) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:19]: In that case the same would apply to the other methods of `M`, as well as wide parts of this PR including the title.

Though I see where this is going; `proto`is currently used mostly for the distinction of network protocols like udp, tcp and icmp in gluon.
Whether this lines up enough with 'fastd', 'babel', 'tunneldigger', 'wireguard' is up for you to decide.

`method` on the other hand, is used less frequently but in the context of (fastd) cypher-methods.

Comment (mweinelt) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:19]: > In that case the same would apply to the other methods of `M`, as well as wide parts of this PR including the title.

Yeah, that was kinda implied. I wasn't going to spam every occurrence, so here we are.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (lemoer) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:33]:
```suggestion
function M.get_active_method()
	-- The "active" method is the (lexically) last of the "available" methods.
```

File: package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/method/tunneldigger.lua
Comment (AiyionPrime) [inline package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/method/tunneldigger.lua:16]: As there are multiple vpn methods, that are available but only one that is returned using `M.get_active_method()` this function might better be called `available()` to avoid a false affordance.
```suggestion
function M.available()
```

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua
Comment (AiyionPrime) [inline package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/method/fastd.lua:17]: Same as in https://github.com/freifunk-gluon/gluon/pull/2102/files#r626032365
```suggestion
function M.available()
```

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (AiyionPrime) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:40]: Related to https://github.com/freifunk-gluon/gluon/pull/2102/files#r626032365
```suggestion
		if method.available() then
```

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (AiyionPrime) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:58]: Related to https://github.com/freifunk-gluon/gluon/pull/2102/files#r626032365
```suggestion
		if not method.available() then
```

**Comment by AiyionPrime** [comment]
@NeoRaider we just saw, you asked for the invert of the changes I proposed;
why do you think is `active` is the better choice in in `package/gluon-mesh-vpn-fastd`and `package/gluon-mesh-vpn-tunneldigger`?

**Comment by lemoer** [comment]
After thinking about this again, I do not get the point of this whole `active()` (or `available()`) function...

If the package `gluon-mesh-vpn-fastd` is selected for the build, the build will fail due to the [check_site.lua script here](https://github.com/freifunk-gluon/gluon/blob/master/package/gluon-mesh-vpn-fastd/check_site.lua#L2) anyway if `mesh_vpn.fastd` is undefined in any of the sites or domains. This means `active()` (or `available()`) would always return true for a valid firmware build, doesn't it? Or did I miss a point here?

**Comment by blocktrron** [comment]
Rationale for `available` is to allow for firmware which packs multiple VPN methods and selects them on per-domain basis.

**Comment by lemoer** [comment]
Ah, ok. Then `check_site.lua` has to be altered, doesn't it?

**Comment by blocktrron** [comment]
Where exactly does a check_site script require to be modified?

**Comment by lemoer** [comment]
[This line](https://github.com/freifunk-gluon/gluon/blob/master/package/gluon-mesh-vpn-fastd/check_site.lua#L2) will fail, if it isn't

**Comment by blocktrron** [comment]
Okay, i see.

The mesh-vpn abstraction is only one step towards the flexible VPN provider goal, so i would go for that in a follow-up PR. This Task is for adapting the integration to the core VPN service, which should initially keep the behavior while trying to achieve an extensible interface we do not shoot ourselves in the foot with.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:33]: Clarified below, added altered description.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:19]: proto is a better fit, thanks for that suggestion.

File: package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/method/tunneldigger.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/method/tunneldigger.lua:16]: As explained to @lemoer , this is intended to have different methods for different domains but one vpn method active per domain. SO I'd stick with active here.

**Comment by blocktrron** [comment]
Adressed the comments.

I'd stick with active in this context, as available would imply a multitude of methods to be available for a site.

File: package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/proto/tunneldigger.lua
Comment (AiyionPrime) [inline package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/proto/tunneldigger.lua:3]: This is a small thing left regarding code-quality;
the require-statements with their used brackets and quotes are somewhat inconsistent.
But other than that the code looks fine now.

File: package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/proto/tunneldigger.lua
Comment (AiyionPrime) [inline package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/proto/tunneldigger.lua:23]: Same nitpick regarding quotes would apply here.

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/proto/fastd.lua
Comment (AiyionPrime) [inline package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/proto/fastd.lua:27]: Why is this swapped?

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (AiyionPrime) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:3]: just a small inconsistency

**Review by AiyionPrime — COMMENTED** [review]
The only thing that matters to me is the question in fastd.lua.
Everything else is just nitpicking.
Looks good.

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/proto/fastd.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/proto/fastd.lua:27]: Not intentional. Fixed.

**Review by AiyionPrime — APPROVED** [review]
The code is still compiling but as the last diff was quite clear, I expect no issues in compiling and running it. The test device will be updated in about 13 minutes.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (lemoer) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:49]:
```suggestion
	for _, name in ipairs(M.get_proto_names()) do
		local proto = M.get_proto(n)

		if proto.active() then
			return name, proto
		end
	end
```

File: package/gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn
Comment (lemoer) [inline package/gluon-mesh-vpn-core/luasrc/lib/gluon/upgrade/500-mesh-vpn:12]: It seems like you missed to replace `'mesh-vpn'` here with your newly introduced method `get_interface()`. I already made another commit, where I touched all the instances of this string. See here: https://github.com/freifunk-gluon/gluon/pull/2168/commits/67cd4e0e07b127557b329df56f143462d3c45658 . You can use this commit to cross check.

File: package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html
Comment (lemoer) [inline package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:9]: I suppose this will break things, if `gluon-mesh-vpn-core` is not baked into the firmware.

**Review by lemoer — COMMENTED** [review]
The rest looks good to me.

File: package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html
Comment (lemoer) [inline package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:15]:
```suggestion
	if has_vpn and vpn.enabled() then
```

File: package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html
Comment (blocktrron) [reply package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:9]: IMHO depending on the core VPN package is better, given it is only pulling in some minified scripts.

Comment (blocktrron) [reply package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:9]: It already depends on `gluon-config-mode-core`

File: package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html
Comment (blocktrron) [reply package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:15]: Depends on config-mode-core.

File: package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html
Comment (lemoer) [reply package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:9]: As of now, this is not a good idea since `gluon-mesh-vpn-core` checks the site.conf with its `check_site.lua`. If we add this dependency, this would mean that even if you want to build an image without vpn, you would have to define e.g. `mesh_vpn.mtu` in your site.conf. (For reference: [Link](https://github.com/freifunk-gluon/gluon/blob/master/package/gluon-mesh-vpn-core/check_site.lua#L2))

Comment (lemoer) [reply package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:9]: `gluon-config-mode-core` != `gluon-mesh-vpn-core`

**Comment by AiyionPrime** [comment]
The current state is built and deployed on

http://meshvpnabstracttest.n.ffh.zone/
2001:678:978:114:b64b:d6ff:fe22:4ebc

**Comment by AiyionPrime** [comment]
Latest push compiles, and runs on http://meshvpnabstracttest.n.ffh.zone/
(2001:678:978:114:b64b:d6ff:fe22:4ebc)

@lemoer `gluon-reconfigure` works without problems.

**Comment by lemoer** [comment]
We have two approvals here. So can we merge this?

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:20]: The term "protocol" seems very inaccurate to me. How about "provider"?

File: package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/proto/fastd.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-fastd/luasrc/usr/lib/lua/gluon/mesh-vpn/proto/fastd.lua:27]: With `simple-uci`, you can use a boolean here.

File: package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/proto/tunneldigger.lua
Comment (NeoRaider) [inline package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/proto/tunneldigger.lua:26]: Tunneldigger also needs simple-tc for upload limit.

**Comment by blocktrron** [comment]
Addressed comments and rebased on #2223

**Comment by AiyionPrime** [comment]
There are currently 17 occurrences of `proto`, are they meant to stay?

**Comment by AiyionPrime** [comment]
Looks better, I'll test later, when the dust has settled a little.

**Comment by blocktrron** [comment]
Works from my testing.

**Comment by AiyionPrime** [comment]
Compiles and gluon-reconfigure does it's job. I'll close blocktrron/gluon#6 then.

File: package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua
Comment (AiyionPrime) [reply package/gluon-mesh-vpn-core/luasrc/usr/lib/lua/gluon/mesh-vpn.lua:27]: Apparently not, nevermind. https://github.com/freifunk-gluon/gluon/blob/a2be901803469d378d6469551a65c20f744aefff/scripts/check_site.lua#L22

**Review by AiyionPrime — CHANGES_REQUESTED** [review]
Just reread the code once more.

File: package/gluon-config-mode-mesh-vpn/luasrc/lib/gluon/config-mode/reboot/0100-mesh-vpn.lua
Comment (blocktrron) [reply package/gluon-config-mode-mesh-vpn/luasrc/lib/gluon/config-mode/reboot/0100-mesh-vpn.lua:21]: Yes

File: package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html
Comment (blocktrron) [reply package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:16]: Yes

File: package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/provider/tunneldigger.lua
Comment (blocktrron) [reply package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/provider/tunneldigger.lua:43]: These methods are now obsolete anyways, as uci::commit is called globally on reconfigure anyways. I removed them now.

File: package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html
Comment (AiyionPrime) [reply package/gluon-web-admin/files/lib/gluon/config-mode/view/admin/info.html:16]: resolved in 98a1c196ed8d76c0449ed85f291b4fac1ace9467

File: package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/provider/tunneldigger.lua
Comment (AiyionPrime) [reply package/gluon-mesh-vpn-tunneldigger/luasrc/lib/lua/gluon/mesh-vpn/provider/tunneldigger.lua]: resolved in 98a1c196ed8d76c0449ed85f291b4fac1ace9467

File: package/gluon-config-mode-mesh-vpn/luasrc/lib/gluon/config-mode/reboot/0100-mesh-vpn.lua
Comment (AiyionPrime) [reply package/gluon-config-mode-mesh-vpn/luasrc/lib/gluon/config-mode/reboot/0100-mesh-vpn.lua]: resolved in 98a1c196ed8d76c0449ed85f291b4fac1ace9467

**Review by AiyionPrime — APPROVED** [review]
Looks good to me.
@NeoRaider or @mweinelt may you review once more?

**Comment by AiyionPrime** [comment]
I compiled the code before the weekend; and just did so again. It's deployed on http://meshvpnabstracttest.n.ffh.zone/cgi-bin/status.
You all still should have access to it.

**Comment by blocktrron** [comment]
Verified initial setup & upgrades are working in a fastd environment.

Let's do this.

**Pull request merged**
